DelvUI/Interface/HudElement.cs

```diff
@@ -1,12 +1,16 @@
 ﻿using Dalamud.Game.ClientState.Actors.Types;
 using DelvUI.Config;
 using System.Numerics;
+using DelvUI.Enums;
 
 namespace DelvUI.Interface
 {
     public abstract class HudElement
     {
         protected MovablePluginConfigObject _config;
+
+        public HudType hudType = HudType.None;
```

Comment:
If you wanna go the enum route, you should make this field _abstract_ and overwrite it in the elements (instead of hardcoding the type for every element in the HudManager).